spark-jobs/src/test/scala/filodb/downsampler/DownsamplerMainSpec.scala

```diff
@@ -490,10 +492,178 @@ class DownsamplerMainSpec extends AnyFunSpec with Matchers with BeforeAndAfterAl
+          |          rules = []
+          |        }
+          |      ]
+          |    }
+          |""".stripMargin
+      ).withFallback(conf)
+      val settings = new DownsamplerSettings(testConf)
+      val rdd = sparkCtx.parallelize(Seq(
+        Row("a", "my_ns"),
+        Row("b", "DNE"),
+        Row("c", "my_ns123"),
+        Row("d", "DNE"),
+      ))
+      val expected = Seq(
+        Row("a", "my_ns"),
+        Row("c", "my_ns123"),
+      )
+      val res = batchExporter.filterRdd(rdd,
+          settings.exportKeyToConfig.head._1,
+          settings.exportKeyToConfig.head._2)
+        .collect().toSeq
+      res shouldEqual expected
+    }
+
+    {
+      val testConf = ConfigFactory.parseString(
+        """
+          |    filodb.downsampler.data-export {
+          |      key-labels = ["_ws_", "_ns_"]
+          |      groups = [
+          |        {
+          |          key = ["_ws_=\"my_ws\"", "_ns_=~\"my_ns.*\""]
+          |          table = "my_ws"
+          |          table-path = "s3a://bucket/directory/catalog/database/my_ws"
+          |          label-column-mapping = [
+          |            "_ws_", "workspace", "NOT NULL",
+          |            "_ns_", "namespace", "NOT NULL"
+          |          ]
+          |          partition-by-columns = []
+          |          rules = []
+          |        }
+          |      ]
+          |    }
+          |""".stripMargin
+      ).withFallback(conf)
+      val settings = new DownsamplerSettings(testConf)
+      val rdd = sparkCtx.parallelize(Seq(
+        Row("my_ws", "my_ns1"),
+        Row("my_ws123", "my_ns2"),
+        Row("c", "c"),
+        Row("my_ws", "DNE"),
+        Row("my_ws", "my_ns3"),
+      ))
+      val expected = Seq(
+        Row("my_ws", "my_ns1"),
+        Row("my_ws", "my_ns3"),
+      )
+      val res = batchExporter.filterRdd(rdd,
+          settings.exportKeyToConfig.head._1,
+          settings.exportKeyToConfig.head._2)
+        .collect().toSeq
+      res shouldEqual expected
+    }
+
+    {
+      val testConf = ConfigFactory.parseString(
+        """
+          |    filodb.downsampler.data-export {
+          |      key-labels = ["_ws_", "_ns_"]
+          |      groups = [
+          |        {
+          |          key = ["_ns_=~\"my_ns.*\"", "_ws_=~\"my_ws.*\""]
+          |          table = "my_ws"
+          |          table-path = "s3a://bucket/directory/catalog/database/my_ws"
+          |          label-column-mapping = [
+          |            "_ws_", "workspace", "NOT NULL",
+          |            "_ns_", "namespace", "NOT NULL"
+          |          ]
+          |          partition-by-columns = []
+          |          rules = []
+          |        }
+          |      ]
+          |    }
+          |""".stripMargin
+      ).withFallback(conf)
+      val settings = new DownsamplerSettings(testConf)
+      val rdd = sparkCtx.parallelize(Seq(
+        Row("my_ws1", "my_ns1"),
```

Comment:
Do we need consider case 

```
Row("my_ws1"),
and
Row("my_ws1", "my_ns1", "another label"),
```

Reply:
Nope -- there will always be as many columns as the labels configured as the "key" above.